src/openforms/submissions/models/submission_report.py

```diff
@@ -65,11 +66,21 @@ def generate_submission_report_pdf(self) -> str:
         :return: string with the HTML used for the PDF generation, so that contents
           can be tested.
         """
+        try:
+            appointment_id = self.submission.appointment_info.appointment_id
```

Comment:
Shouldn't we check if the `appointment_info` exists first and then access the `appointment_id`?

Reply:
`.` operator is left associative, meaning it's evaluated from left to right like `((self.submission).appointment_info).appointment_id`. So at the second braces `Submission.__get_attribute__("appointment_info")` (or `__getattr__`) will raise `AppointmentInfo.DoesNotExist` (not an `AttributeError` 🙹) before it gets to the last `.`, which is explicitly caught with an `except` block.

If you want to learn more about operator overloading in Python, [Expecter Gadget](https://github.com/garybernhardt/expecter) is a more approachable code base. Or https://github.com/garybernhardt/dingus, which I was using before unittest.mock was in the standard library,